### PR TITLE
Make Pay trait from salaries pallet more generic

### PR DESCRIPTION
**Pull request by franciscoaguirre**
- Moved the `Pay` trait to live in `frame_support`.
- Added a new associated type `AssetKind` to enable usage of this trait via XCM (and possibly others?).
- Changed `AccountId` to `Beneficiary` since we may want to pay more general concepts like XCM's `MultiLocation`
- Used unit type for `AssetKind` on the salary pallet

Solves https://github.com/paritytech/substrate/issues/13603.

### CODE REVIEW

**Review by gavofyork — APPROVED** [review]

File: frame/support/src/traits/tokens/pay.rs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	/// Make a payment and return an identifier for later evaluation of success in some off-chain
+	/// mechanism (likely an event, but possibly not on this chain).
+	fn pay(
+		who: &Self::Beneficiary,
+		asset_kind: Self::AssetKind,
+		amount: Self::Balance,
+	) -> Result<Self::Id, ()>;
+	/// Check how a payment has proceeded. `id` must have been previously returned by `pay` for
+	/// the result of this call to be meaningful.
+	fn check_payment(id: Self::Id) -> PaymentStatus;
```
Comment (gavofyork) [inline frame/support/src/traits/tokens/pay.rs:47]:
```suggestion
	/// Check how a payment has proceeded. `id` must have been previously returned by `pay` for
	/// the result of this call to be meaningful. Once this returns anything other than
	/// `InProgress` for some `id`, the result of further calls with the same `id` is
	/// not defined.
	fn check_payment(id: Self::Id) -> PaymentStatus;
```

**Review by tonyalaribe — APPROVED** [review]

**Review by joepetrowski — APPROVED** [review]

**Comment by franciscoaguirre** [comment]
bot rebase

**Comment by franciscoaguirre** [comment]
bot merge

**Comment by franciscoaguirre** [comment]
bot rebase

**Review by ggwpez — APPROVED** [review]

**Comment by franciscoaguirre** [comment]
bot rebase

**Comment by franciscoaguirre** [comment]
bot merge

**Pull request merged**